### PR TITLE
Break out features into separate files

### PR DESCRIPTION
**Pull request by incompl**

### CODE REVIEW

File: src/redispatch.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+expect despite the presense of ads. For example, an integrator would expect
+an `ended` event to happen once the content is ended. If an `ended` event is sent
+as a result of an ad ending, that is a bug. The `redispatch` method should recognize
+such `ended` events and prefix them so they are sent as `adended`, and so on with
+all other player events.
+*/
+
+import videojs from 'video.js';
+
+let player;
```
Comment (gkatsev) [inline src/redispatch.js]: this would get shared between all players.

Comment (gkatsev) [inline src/redispatch.js]: Looking at the code, it *might* be ok as is but we should probably try to get rid of this if we can.

File: src/videojs.ads.css [hunk 2]
```diff
@@ -1,16 +0,0 @@
-/**
- * videojs.ads.css
```
Comment (gkatsev) [inline src/videojs.ads.css]: why is this file removed?

Comment (incompl) [inline src/videojs.ads.css]: Moved to plugin.scss

File: src/redispatch.js (hunk 1 above)
Comment (incompl) [inline src/redispatch.js]: I have been thinking about the best way to do this. One option is to make this a class and you pass a player to the constructor. What do you think?

File: src/videojs.ads.css (hunk 2 above)
Comment (gkatsev) [inline src/videojs.ads.css]: Oh, I see. Missed it from the other PR.

File: src/redispatch.js (hunk 1 above)
Comment (gkatsev) [inline src/redispatch.js]: Is the only reason we're using `player` here for `prefixEvent`? What about passing `player` through to it directly.

Comment (incompl) [inline src/redispatch.js]: It ends up being a lot of boilerplate that I want to avoid. Of course, a class is a lot of boilerplate too.

Comment (gkatsev) [inline src/redispatch.js]: Maybe wrap it all in a function?

Comment (incompl) [inline src/redispatch.js]: Just ended up passing player directly, at least it's explicit and clear

File: src/cancelContentPlay.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/*
+This feature makes sure the player is paused during ad loading.
+
+It does this by pausing the player immediately after a "play" where ads will be requested,
+then playing after the ad is done if we paused the player in this way.
```
Comment (ldayananda) [inline src/cancelContentPlay.js]: This technically doesn't unpause the player in this file

**Comment by bcvio** [comment]
LGTM
